WooCommerce/Classes/ViewRelated/Hub Menu/HubMenuViewModel.swift

```diff
@@ -277,23 +300,42 @@ final class HubMenuViewModel: ObservableObject {
             }
             .assign(to: &$shouldAuthenticateAdminPage)
 
-        // Blaze menu.
         stores.site
             .compactMap { $0 }
-            .removeDuplicates()
-            .asyncMap { [weak self] site -> Bool in
-                guard let self else {
-                    return false
-                }
-                return await self.blazeEligibilityChecker.isSiteEligible()
+            .filter { [weak self] site in
+                site.siteID == self?.siteID
```

Comment:
❓I couldn't understand why we have to check the siteID here and not at other places where we subscribe to `stores.site`. 🤔  Kindly consider adding a comment to explain why we have this check.

Reply:
When switching sites, the view model is recreated with a new site ID but the site info needs more time to be fetched and updated in the store manager. That's why the initial element emitted here would be the previous site's info. This filter helps avoid fetching data for the previous site.

I added the comment and reuse the same value for other subscriptions here d63e244cd673642a7c938c7ec6c460726175563d.